src/methods/michelsen/tpd.jl

```diff
@@ -0,0 +1,259 @@
+
+
+
+function tpd_obj!(model::EoSModel, p, T, di, α, phasew, z_notzero; volw0=nothing,
+                  F=nothing, G=nothing, H=nothing)
+    # Function that computes the TPD function, its gradient and its hessian
+    nc = length(model)
+    ncomponents = length(α)
+
+    w = zeros(nc)
+    w[z_notzero] = α.^2 /4.
+    w_notzero = w[z_notzero]
```

Comment:
instead of making a new vector that is a subset of the non-zero values and evaluating this in each iteration, we have dedicated functionality for that: given a `model::EoSModel` and `z::AbstractVector`, you can obtain `model_reduced,z_reduced = index_reduction(model,z)` such as `helmholtz(model,v,T,z) = helmholtz(model_reduced,V,T,z_reduced)`. by reducing the model size, you can evaluate less and be faster